File: backend/src/models/Draft.js
import { Schema, model } from 'mongoose';

const draftSchema = new Schema(
  {
    // Author
    firstName:   { type: String, required: true },
    lastName:    { type: String, required: true },
    email:       { type: String, required: true },
    studentId:   { type: String, required: true },
    affiliation: { type: String, default: 'American University of Beirut' },
    year:        String,
    major:       String,

    // Article
    title:    { type: String, required: true },
    abstract: { type: String, required: true },
    category: { type: String, enum: ['cmps', 'math', 'phys'], required: true },
    keywords: [String],

    // File
    pdfPath:  String,
    fileName: String,
    fileSize: Number,

    // Metadata
    isOriginalWork:   Boolean,
    hasEthicsApproval:Boolean,
    agreedToTerms:    Boolean,
    status: { type: String, enum: ['pending', 'approved', 'rejected'], default: 'pending' }
  },
  { timestamps: true }
);

export default model('Draft', draftSchema);

File: backend/src/main/java/com/eigenjournal/model/Draft.java
package com.eigenjournal.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import lombok.Data;

@Data                           // adds getters, setters, toString, etc.
@Entity
@Table(name = "drafts")
public class Draft {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // --- author info ---
    private String firstName;
    private String lastName;
    private String email;
    private String studentId;
    private String affiliation;
    private String year;
    private String major;

    // --- article info ---
    private String title;
    @Column(length = 2000)   // long abstracts
    private String abstractText;
    private String category;         // cmps / math / phys
    private String keywords;         // comma‑separated

    // --- file & meta ---
    private String pdfFileName;      // stored filename
    private LocalDateTime createdAt;
    private String status;           // PENDING_REVIEW

    /* getters and setters omitted for brevity;
       using Lombok. */
}
